kernelci/build.py

```diff
@@ -787,8 +787,6 @@ def _output_to_file(self, cmd, file_path, rel_path=None):
     def _get_make_opts(self, opts, make_path):
         env = self._meta.get('bmeta', 'environment')
         make_opts = env['make_opts'].copy()
-        if opts:
-            make_opts.update(opts)
```

Comment:
Maybe I'm missing something, but why you're removing this?

Reply:
It's moved to below (L820). So we can overwrite O= option.
The reason is, it seems that is we assign O=/some/path/other/than/make_path, the compile will fail. And set O to makefile path is fine. The error:
In file included from bench/../../arch/x86/lib/memcpy_64.S:8,
                 from bench/mem-memcpy-x86-64-asm.S:14:
/root/kernelci-core/linux/build/arch/x86/include/generated/asm/export.h:1:10: fatal error: asm-generic/export.h: No such file or directory
 #include <asm-generic/export.h>
          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.

The full step:
./kci_build make_config --defconfig=defconfig+kselftest
./kci_build make_kernel
./kci_build make_perftool

Reply:
OK this shouldn't be breaking existing use-cases where the `O=` option is not defined for specific builds, and it gives the possibility to override it when needed.  We'll see if all the builds work as expected on staging to confirm this.

Reply:
It took about 20s to build on my regular builder with 4 cores, and I believe this is the only artifact we need to store:
```
-rwxr-xr-x 1 root root 14M Jun  2 14:01 tools/perf/perf
```
And after running `strip` on it:
```
-rwxr-xr-x 1 root root 5.7M Jun  2 14:04 tools/perf/perf
```
So I think we can try to just enable it for all the main defconfigs, it's not that big.  Maybe we can just build it for the main defconfig of each architecture if we can configure this easily in YAML.